CRM/Contribute/BAO/Contribution.php

```diff
@@ -2789,6 +2791,8 @@ public function _gatherMessageValues($input, &$values, $ids = array()) {
     }
     $values['customGroup'] = $customGroup;
 
+    $values['is_pay_later'] = $this->is_pay_later;
```

Comment:
I think this should be better done at a place where all contribution page related values are assigned (addContributionPageValuesToValuesHeavyHandedly() function). Can you please try adding `'is_pay_later'` at this point https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/BAO/Contribution.php#L5056 and see if it works for you?